### ISSUE TITLE
Assert.Pass(message) produces "reason" in XML-Result

### ISSUE THREAD

**Issue by sja-schleupen** [nunit/nunit#3643]
Using Assert.Pass(message) in a successful unittest results in a reason-element in XML.

This result seems suprising. Jenkins interprets "reason" as a skipped test in all cases:
https://github.com/jenkinsci/nunit-plugin/blob/master/src/main/resources/hudson/plugins/nunit/nunit-to-junit.xsl

According to https://docs.nunit.org/articles/nunit/technical-notes/usage/Test-Result-XML-Format.html#reason:
Reason is an "Optional element that may appear on tests or suites that were not executed. Contains a message giving the reason for skipping the test."

Cause seems to be:
https://github.com/nunit/nunit/blob/ad49f27294bd0f2677d8699756c6ccb10df600f8/src/NUnitFramework/framework/Internal/Results/TestResult.cs#L385-L400

**Steps to reproduce**
NUnit-Version: 3.12
NUnit.ConsoleRunner-Version: 3.11.1

Sample unittest:
```
[Test]
public void PassWithMessage()
{
  Assert.Pass("Sample");
}
```

XML-Output
```
<test-case id="0-1002" name="PassWithMessage" fullname="Sample.Test.PassWithMessage" methodname="PassWithMessage" classname="Sample.Test" runstate="Runnable" seed="1930288149" result="Passed" start-time="2020-10-07 06:50:32Z" end-time="2020-10-07 06:50:33Z" duration="0.103248" asserts="0">
    <reason>
        <message><![CDATA[Sample]]></message>
    </reason>
</test-case>
```

**Comment by mikkelbu** [comment]
I would say that this is an error on the documentation page. The documentation of `Assert.Pass(message)` states

> Throws a NUnit.Framework.SuccessException with the message and arguments that are passed in. This allows a test to be cut short, with a result of success returned to NUnit.

So I would say that the documentation page should be updated, so that it mentions that the `reason` can both be the reason for not executing the test or the reason for short circuiting the test. And then the Jenkins NUnit plugin needs to be updated (it can e.g. examine the `result` attribute on `test-case` to see if the test was skipped - then it will have the value `Skipped`).

**Comment by mikkelbu** [comment]
Closed by nunit/docs#584
